manifest.go

```diff
@@ -117,6 +121,9 @@ func validateManifest(s string) ([]error, error) {
 								warns = append(warns, fmt.Errorf("Invalid key %q in %q", key, prop))
 							}
 						}
+						if !ruleProvided && len(props) > 0 {
```

Comment:
Let's warn for no `name` attribute in constraint/override too. With that we can be sure, we have at least one attribute and not empty. And this check should be performed before checking for missing rule.

Also, since there won't be any changing part in missing `name` in constraint/override warning, you can create 2 separate variables for them, similar to the error variables in https://github.com/golang/dep/blob/v0.3.2/manifest.go#L31 . And use the same variables in tests to perform comparison.